mono/metadata/assembly.c

```diff
@@ -1669,6 +1672,13 @@ netcore_load_reference (MonoAssemblyName *aname, MonoAssemblyLoadContext *alc, M
 	if (reference)
 		goto leave;
 
+	// See: https://github.com/dotnet/coreclr/blob/0a762eb2f3a299489c459da1ddeb69e042008f07/src/vm/appdomain.cpp#L5178-L5239
+	if (!(strcmp (aname->name, MONO_ASSEMBLY_CORLIB_NAME) == 0 && is_satellite) && postload) {
```

Comment:
I'm still unsure on under what conditions postload should be set or not, but maybe it's best to get this merged and fix issues as they come up about it being run when it shouldn't? I'll take another look at CoreCLR tomorrow and see if it's clearer to me.